### PR TITLE
Issue 17594 future time machine not showing working content

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**
The whole issue with time-machine is related to the recent changes that prevent anonymous users from being able to view **working** content. The variable `$user` in the vtl-code was always null.
which basically translates into anonymous
That didn't seem like an issue for published content. But it breaks the time-machine code.
So.. long story short initializing the user with the current logged in user seems to solve the problem.
I need to validate with QA that the behavior seen using the log-in-as functionality is what is expected here. It does seem so.

Additionally to that, the ResourceManager had its own private cache manager. One that was disconnected from the one provided by the CacheLocator. I changed that so when you invalidate the velocity cache from the UI. These templates are also invalidated. 

And last but not least I'm cleaning the `tm_date` session variable on the PageResource since the resulting page-rendered might conflict with that value which is supposed to only be set for the time machine.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/services/DotLoader.java [hunk 1]
```diff
@@ -61,20 +57,24 @@ default InputStream writeOutVelocity(final String filePath, final String strOut)
                 File f = new File(ConfigUtils.getDynamicVelocityPath() + java.io.File.separator + filePath);
                 f.mkdirs();
                 f.delete();
-                try (final VelocityPrettyWriter out = new VelocityPrettyWriter(new FileOutputStream(f))) {
+                try (final VelocityPrettyWriter out = new VelocityPrettyWriter(System.out)) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/services/DotLoader.java]: Codacy Issue found: [Unnecessary modifier 'final' on resource specification 'out': resource specifications are implicitly final](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4504016)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java [hunk 2]
```diff
@@ -186,6 +186,10 @@ private InputStream buildVelocity(final Container container, final String uuid,
             .append(container.getInode())
             .append("')");
 
+        velocityCodeBuilder.append("#if(!$UtilMethods.isSet($user)) ")
+        .append("#set($user = $cmsuser.getLoggedInUser($request)) ")
+        .append("#end");
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java]: Codacy Issue found: [The String literal "#end" appears 14 times in this file; the first occurrence is on line 108](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4504016)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java [hunk 3]
```diff
@@ -331,31 +337,41 @@ private InputStream buildVelocity(final Container container, final String uuid,
...
-            
+
                             for (int i = 0; i < containerContentTypeList.size(); i++) {
                                 ContainerStructure cs = containerContentTypeList.get(i);
                                 String ifelse = (i == 0) ? "if" : "elseif";
                                 velocityCodeBuilder.append("#" + ifelse + "($ContentletStructure ==\"" + cs.getStructureId() + "\")");
                                 velocityCodeBuilder.append(cs.getCode());
                             }
-                            if (containerContentTypeList.size() > 0) velocityCodeBuilder.append("#end");
+                            if (containerContentTypeList.size() > 0) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java]: Codacy Issue found: [Substitute calls to size() == 0 (or size() != 0, size() > 0, size() < 1) with calls to isEmpty()](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4504016)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java]: Codacy Issue found: [The String literal "#end" appears 14 times in this file; the first occurrence is on line 108](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4504016)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java]: Codacy Issue found: [Substitute calls to size() == 0 (or size() != 0, size() > 0, size() < 1) with calls to isEmpty()](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4504016)

**Review by jdotcms — APPROVED** [review]
good

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java [hunk 4]
```diff
@@ -331,31 +338,41 @@ private InputStream buildVelocity(final Container container, final String uuid,
...
                         velocityCodeBuilder.append("#if($isWidget==true)").append("$widgetCode");
                         velocityCodeBuilder.append("#elseif($isForm==true)").append("$formCode");
                         velocityCodeBuilder.append("#else");
-            
+
                             for (int i = 0; i < containerContentTypeList.size(); i++) {
                                 ContainerStructure cs = containerContentTypeList.get(i);
                                 String ifelse = (i == 0) ? "if" : "elseif";
-                                velocityCodeBuilder.append("#" + ifelse + "($ContentletStructure ==\"" + cs.getStructureId() + "\")");
+                                velocityCodeBuilder.append("#").append(ifelse)
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java]: Codacy Issue found: [Avoid appending characters as strings in StringBuffer.append.](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4504016)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java [hunk 5]
```diff
@@ -331,31 +338,41 @@ private InputStream buildVelocity(final Container container, final String uuid,
...
+                                        .append("($ContentletStructure ==\"")
+                                        .append(cs.getStructureId()).append("\")");
                                 velocityCodeBuilder.append(cs.getCode());
                             }
-                            if (containerContentTypeList.size() > 0) velocityCodeBuilder.append("#end");
+                            if (containerContentTypeList.size() > 0) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java]: Codacy Issue found: [Substitute calls to size() == 0 (or size() != 0, size() > 0, size() < 1) with calls to isEmpty()](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4504016)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java]: Codacy Issue found: [Avoid appending characters as strings in StringBuffer.append.](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4504016)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java]: Codacy Issue found: [Substitute calls to size() == 0 (or size() != 0, size() > 0, size() < 1) with calls to isEmpty()](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4504016)

**Review by jgambarios — APPROVED** [review]

**Pull request merged**